### PR TITLE
Downgrade insight

### PR DESCRIPTION
**Pull request by jburel**
In this PR, review the way we read entries from the catalog
To test:
 * Select an image. 
 * Select to export it as ome-tiff
 * Check that the menu with older versions is displayed
 * Select an older version and click export.
 * Re-import the generated ome-tiff

This should now work on mac windows and linux
Check if the option is also available when using webstart

### CODE REVIEW

**Comment by dominikl** [comment]
Works fine on Mac (standalone and webstart), Linux and Windows. But exporting to an older ome.tiff on Windows claims to be failed with the exception seen below, but it actually worked (the "failed" file looks ok, and can be re-imported without problems)
```
java.lang.Exception: java.lang.IllegalArgumentException: Unable to apply the transforms
	at org.openmicroscopy.shoola.env.data.OmeroImageServiceImpl.exportImageAsOMEFormat(OmeroImageServiceImpl.java:1515)
	at org.openmicroscopy.shoola.env.data.views.calls.ExportLoader$1.doCall(ExportLoader.java:87)
	at org.openmicroscopy.shoola.env.data.views.BatchCall.doStep(BatchCall.java:144)
	at org.openmicroscopy.shoola.util.concur.tasks.CompositeTask.doStep(CompositeTask.java:226)
	at org.openmicroscopy.shoola.env.data.views.CompositeBatchCall.doStep(CompositeBatchCall.java:126)
	at org.openmicroscopy.shoola.util.concur.tasks.ExecCommand.exec(ExecCommand.java:165)
	at org.openmicroscopy.shoola.util.concur.tasks.ExecCommand.run(ExecCommand.java:276)
	at org.openmicroscopy.shoola.util.concur.tasks.AsyncProcessor$Runner.run(AsyncProcessor.java:91)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.Exception: Cannot apply transform
	at org.openmicroscopy.shoola.env.data.OmeroImageServiceImpl.applyTransforms(OmeroImageServiceImpl.java:1440)
	at org.openmicroscopy.shoola.env.data.OmeroImageServiceImpl.exportImageAsOMEFormat(OmeroImageServiceImpl.java:1498)
	... 8 more
Caused by: java.lang.NullPointerException
	at org.openmicroscopy.shoola.env.data.OmeroImageServiceImpl.applyTransforms(OmeroImageServiceImpl.java:1427)
	... 9 more
```
As I had problems with my Parallels/Windows8.1 setup before, which couldn't be reproduced by anyone else, maybe @pwalczysko could have quick test on Windows just to be sure.

**Comment by jburel** [comment]
@dominikl: not related to Parallels, I can reproduce the issue on Windows

**Comment by jburel** [comment]
@dominikl: problem should be fixed.

**Comment by dominikl** [comment]
Mac and Linux working fine, like before. I can export on Windows without error message, but if I re-import the images exported to an older ome.tiff version on Windows, the channels are missing and there's just a black thumbnail. See 'read-only-1/user-3/downgrade_test_reimport_win' (compared to 'downgrade_test_reimport_mac' which worked fine), original image in dataset 'downgrade_test'.

**Comment by jburel** [comment]
@dominikl: not all information will be preserve depending on the schema you use.
e.g. using 2003-FC implies that a lot of metadata will be removed.

**Comment by dominikl** [comment]
Must be Windows export specific. Same ome.tiff version exported and reimported on Mac works fine. But a reimport on Mac with the images exported on Windows shows the same problem.

**Comment by jburel** [comment]
interesting, I think this is an encoding issue. see https://github.com/jburel/openmicroscopy/commit/6afe4dd30150632d291fb216e4cf3c9a4bb9583d

**Comment by jburel** [comment]
@dominikl: the last commit should fix the issue. okay to retest this afternoon with new build?

**Comment by dominikl** [comment]
Now it also works on Windows! Good to merge.

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OmeroImageServiceImpl.java [hunk 1]
```diff
@@ -1386,7 +1392,61 @@ public ScriptCallback createMovie(SecurityContext ctx, long imageID,
...
+                inputXML = output;
+                stream.close();
+                out.close();
+                in.close();
+                resolver.close();
+            }
+        } catch (Exception e) {
+            throw new Exception("Cannot apply transform", e);
+        }
+        File f = File.createTempFile(
```
Comment (joshmoore) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OmeroImageServiceImpl.java]: Should this be deleteOnExit as well?

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OmeroImageServiceImpl.java [hunk 2]
```diff
@@ -1386,7 +1392,61 @@ public ScriptCallback createMovie(SecurityContext ctx, long imageID,
...
+                output = File.createTempFile(
+                        RandomStringUtils.random(60, false, true),"."+XMLFilter.OME_XML);
+                output.deleteOnExit();
+                Source src = new StreamSource(stream);
+                Templates template = factory.newTemplates(src);
+                transformer = template.newTransformer();
+                transformer.setParameter(OutputKeys.ENCODING, encoding);
+                out = new FileOutputStream(output);
+                in = new FileInputStream(inputXML);
+                transformer.transform(new StreamSource(in),
```
Comment (joshmoore) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OmeroImageServiceImpl.java]: If this throws, will the streams' being left open cause a problem?

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OmeroImageServiceImpl.java (hunk 1 above)
Comment (jburel) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OmeroImageServiceImpl.java]: file is deleted in the finally.
I could certainly add a try catch finally in the loop to make sure  it is closed

**Comment by jburel** [comment]
streams should now be closed.

**Comment by dominikl** [comment]
Looks fine to me; ome.tiff export still works as expected (re-tested on Mac and Windows).

**Comment by joshmoore** [comment]
Thanks, guys.

**Comment by jburel** [comment]
--no-rebase

**Pull request merged**